stl/src/xonce2.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// __std_execute_once_begin & __std_execute_once_end functions
+
+#include "xcall_once.h"
+#include <atomic>
+#include <libloaderapi.h>
+#include <synchapi.h>
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+// these declarations must be in sync with those in xcall_once.h
+
+#if _STL_WIN32_WINNT >= _WIN32_WINNT_VISTA
+
+_STD_BEGIN
+int __CLRCALL_PURE_OR_CDECL __std_execute_once_begin(
+    once_flag& _Once_flag, int& _Pending, bool& fallback) noexcept { // wrap Win32 InitOnceBeginInitialize()
+    static_assert(sizeof(_Once_flag._Opaque) == sizeof(INIT_ONCE), "invalid size");

```

Comment:
You repeat this within 4 functions. I think a single

```cpp
static_assert(sizeof(std::once_flag::_Opaque) == sizeof(INIT_ONCE), "invalid size");
```

at global scope, outside of the "if Vista" logic, would be fine.